lib/puppet/type/archive.rb

```diff
@@ -103,7 +103,7 @@ def should_to_s(value)
   newparam(:source) do
     desc 'archive file source, supports http|https|ftp|file uri.'
     validate do |value|
-      unless value =~ URI.regexp(%w(http https file ftp))
+      unless value =~ URI.regexp(%w(http https file ftp)) || value =~ %r{^([A-Z]:|\/)}
```

Comment:
I meant I run into problems testing windows filepath in rspec if I use Puppet::Util.absolute_path?

```puppet
  1) Puppet::Type::Archive verify resoource[:source] is valid source
     Failure/Error: expect do
       expected no Exception, got #<Puppet::ResourceError: Parameter source failed on Archive[/tmp/example.zip]: invalid source url: D:/example.zip> with backtrace:
         # ./.bundle/gems/puppet-3.8.4/lib/puppet/util/errors.rb:106:in `fail'
         # ./lib/puppet/type/archive.rb:107:in `block (3 levels) in <top (required)>'
         # ./.bundle/gems/puppet-3.8.4/lib/puppet/parameter.rb:465:in `validate'
         # ./.bundle/gems/puppet-3.8.4/lib/puppet/parameter.rb:500:in `value='
         # ./.bundle/gems/puppet-3.8.4/lib/puppet/type.rb:654:in `[]='
         # ./spec/unit/puppet/type/archive_spec.rb:35:in `block (3 levels) in <top (required)>'
         # ./spec/unit/puppet/type/archive_spec.rb:30:in `block (2 levels) in <top (required)>'
     # ./spec/unit/puppet/type/archive_spec.rb:30:in `block (2 levels) in <top (required)>'
```

The spec behavior is different depending on where you run it. On Mac/Linux it expects / path, and Windows [A-Z]: path. I'm looking for a proper way to stub platform detection for this method, otherwise rake spec will fail.

I suppose as last resort I can get rid of the test cases for the file path and just trust it does the right thing.